// Copyright 2021 The XLS Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package xlscc;

enum ChannelType {
  INVALID = 0;
  DIRECT_IN = 1;
  FIFO = 2;
  MEMORY = 3;
}

message HLSChannel {
  optional string name = 1;
  optional bool is_input = 2;
  optional ChannelType type = 3;
  optional int32 width_in_bits = 4;
  optional int32 depth = 5;  // For memories
}

message HLSBlock {
  repeated HLSChannel channels = 1;
  optional string name = 2;
}
